src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -136,46 +136,29 @@ class ArtifactRequirement:
     @classmethod
     def from_jvm_artifact_target(cls, target: Target) -> ArtifactRequirement:
         if not JvmArtifactFieldSet.is_applicable(target):
-            raise CoursierError(
+            raise AssertionError(
                 "`ArtifactRequirement.from_jvm_artifact_target()` only works on targets with "
                 "`JvmArtifactFieldSet` fields present."
             )
 
-        group = target[JvmArtifactGroupField].value
-        if not group:
-            raise InvalidTargetException(
-                f"The `group` field of {target.alias} target {target.address} must be set."
-            )
-
-        artifact = target[JvmArtifactArtifactField].value
-        if not artifact:
-            raise InvalidTargetException(
-                f"The `artifact` field of {target.alias} target {target.address} must be set."
-            )
-
-        version = target[JvmArtifactVersionField].value
-        if not version:
-            raise InvalidTargetException(
-                f"The `version` field of {target.alias} target {target.address} must be set."
-            )
```

Comment:
The Target API already was checking that it's not `None`. We just needed to teach MyPy by setting `value: str`, as MyPy doesn't understand that `required = True` means the `value` is no longer `Optional`.